pkg/metrics/metrics.go

```diff
@@ -55,7 +55,7 @@ func CreateMetricsService(ctx context.Context, cfg *rest.Config, servicePorts []
 	if err != nil {
 		if err == k8sutil.ErrNoNamespace || err == k8sutil.ErrRunLocal {
 			log.Info("Skipping metrics Service creation; not running in a cluster.")
-			return nil, nil
+			return nil, err
```

Comment:
This is not the change I was implying. It would be better to leave this as-is and remove the above checks.